server/src/main/java/org/spine3/server/entity/storage/Column.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.entity.storage;
+
+import org.spine3.server.entity.Entity;
+
+import javax.annotation.Nullable;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.lang.String.format;
+
+/**
+ * The representation of a field of an {@link Entity} which is stored in the storage in the way
+ * efficient for querying.
+ *
+ * <p>A Column is a value retrieved from an {@link Entity} getter.
+ *
+ * <h2>Type policy</h2>
+ *
+ * <p>A Column can turn into any type. If use a ready implementation of
+ * the {@link org.spine3.server.storage.Storage Spine Storages}, the most commonly used types should
+ * be already supported. However, you may override the behavior for any type whenever you  wish. For
+ * more info, see {@link ColumnTypeRegistry}.
+ *
+ * <p>To handle specific types of the Columns, which are not supported by default,
+ * implement the {@link ColumnType} {@code interface}, register it in a {@link ColumnTypeRegistry}
+ * and pass the instance of the registry into the
+ * {@link org.spine3.server.storage.StorageFactory StorageFactory} on creation.
+ *
+ * <h2>Nullability</h2>
+ *
+ * <p>A Column may turn into {@code null} value if the getter which declares it is annotated as
+ * {@link javax.annotation.Nullable javax.annotation.Nullable}. Otherwise, the Column is considered
+ * non-null.
+ *
+ * <p>If a non-null getter method returns {@code null} when trying to get the value of a Column,
+ * a {@linkplain RuntimeException} is thrown. See {@link #isNullable()}.
+ *
+ * @see ColumnType
+ * @author Dmytro Dashenkov
+ */
+public class Column<T> {
```

Comment:
Some of the code samples that you have listed in the PR description would look nice here as well.